### PR TITLE
first round of challanges

### PR DESCRIPTION
**Pull request by heyitsgeo**
Here is the first round of challenge files. I am going to write a separate document for what I am looking for. The things that are obvious issues should be obvious. However, if something looks like an actually mistake on my part let me know :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,5 @@
 # Peer Programming Challanges
+Use this project as a way to challange developers with relativly real world examples related to RESTFUl APIs and frontend technologies. By loading this repository via [https://vscode.dev/github/heyitsgeo/peer-programming-challanges](https://vscode.dev/github/heyitsgeo/peer-programming-challanges) developers can edit code in a browser in without needing to install or clone anything on thier machine.
```
Comment (paullhmiller) [inline README.md]: I think you want "RESTful" instead of "RESTFUL"

Comment (paullhmiller) [reply README.md]: "on their machine."

Comment (paullhmiller) [reply README.md]: "developers can edit code in a browser without needing"

File: c#/.net-controller-challange/ISubmissionsService.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+public interface ISubmissionsService
+{
+    Task<IList<Submission>> GetPendingSubmissions(Datetime from, Datetime to, CancellationToken);
```
Comment (paullhmiller) [inline c#/.net-controller-challange/ISubmissionsService.cs]: CancellationToken needs a param name

Comment (paullhmiller) [reply c#/.net-controller-challange/ISubmissionsService.cs]: And Datetime should be DateTime

File: c#/.net-controller-challange/Submission.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+public class Submission
+{
+    public int Id { get; set; }
+
+    public string SubmissionNumber { get ; set; }
+
+    public Product Product { get; set; } 
+
+    public Datetime CreatedDate { get; set; }
```
Comment (paullhmiller) [inline c#/.net-controller-challange/Submission.cs]: DateTime

File: c#/.net-controller-challange/SubmissionsController.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+{
+    private readonly ISubmissionsService _submissionsService;
+
+    public SubmissionsController(ISubmissionsService submissionsService) {
+        _submissionsService = submissionsService;
+    }
+
+    // TODO: submissions/pending?from=03-03-2022&to=03-31-2022
+
+    [Route({id})]
```
Comment (paullhmiller) [inline c#/.net-controller-challange/SubmissionsController.cs]: I think you want double quotes here

File: c#/.net-controller-challange/SubmissionsController.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    public async Task<IActionResult> UpdateSubmission(
+        [FromRoute] int id,
+        [FromBody] UpdateSubmissionRequest request,
+        CancellationToken cancellationToken,
+    )
+    {
+        var submission = // TODO: get a submission by Id
+
+        await _submissionsService.UpdateSubmission(submission);
```
Comment (paullhmiller) [inline c#/.net-controller-challange/SubmissionsController.cs]: missing cancellationToken

File: c#/.net-controller-challange/SubmissionsController.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
+[Route("submissions")]
+public class SubmissionsController : ControllerBase
+{
+    private readonly ISubmissionsService _submissionsService;
+
+    public SubmissionsController(ISubmissionsService submissionsService) {
```
Comment (paullhmiller) [inline c#/.net-controller-challange/SubmissionsController.cs]: Opening brace is on a new line in C#, right?

File: c#/.net-controller-challange/SubmissionsService.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
+public class SubmissionsService : ISubmissionsService
+{
+    private readonly DbContext _dbContext;
+
+    public SubmissionsService(DbContext dbContext) {
+        _dbContext = dbContext;
+    }
+
+    public async Task<IList<Submission>> GetPendingSubmissions(Datetime from, Datetime to, CancellationToken) {
```
Comment (paullhmiller) [inline c#/.net-controller-challange/SubmissionsService.cs]: DateTime case

CancellationToken param name

File: c#/.net-controller-challange/SubmissionsService.cs [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+        // TODO: Add filters
+        return await _dbContext.Set<Submission>().ToListAsync();
```
Comment (paullhmiller) [inline c#/.net-controller-challange/SubmissionsService.cs]: `ToListAsync(cancellationToken)`

File: js/http-utility-method-challange/vue-component-challange/submissions-page.vue [hunk 9]
```diff
@@ -0,0 +1,84 @@
+<template>
+    <div class"header">
+        <h1 class="header__title">Submissions</h1>
+        <h1 class="header__subheading">See all submissions, edit them, and add new ones.</h1>
+    </div>
+    <div v-if="!editingSubmission || !creatingSubmission">
+        <ul v-if="submissions.length > 0">
+            <li v-for="submission in submissions" :key="submission.id" @click>
```
Comment (paullhmiller) [inline js/http-utility-method-challange/vue-component-challange/submissions-page.vue]: empty click

**Review by paullhmiller — COMMENTED** [review]
Just ignore any of my comments that are pointing out intentional defects :)

File: c#/.net-controller-challange/SubmissionsController.cs (hunk 5 above)
Comment (heyitsgeo) [reply c#/.net-controller-challange/SubmissionsController.cs]: This is intentional :)

File: c#/.net-controller-challange/SubmissionsService.cs (hunk 8 above)
Comment (heyitsgeo) [reply c#/.net-controller-challange/SubmissionsService.cs]: This is intentional.

**Comment by stevenbenitez** [comment]
I would recommend changing the folders for the .NET challenge from `/c#/.net-controller-challange/` to `/csharp/dotnet-controller-challenge/`.

I'm not sure whether the `#` character will cause any problems on any operating systems. Also, folders starting with `.` may be hidden by some operating systems.

**Comment by stevenbenitez** [comment]
The word challenge is misspelled as challange throughout this repo, including folder names. :)

File: README.md [hunk 10]
```diff
@@ -1 +1,5 @@
 # Peer Programming Challanges
+Use this project as a way to challange developers with relativly real world examples related to RESTful APIs and frontend technologies. By loading this repository via [https://vscode.dev/github/heyitsgeo/peer-programming-challanges](https://vscode.dev/github/heyitsgeo/peer-programming-challanges) developers can edit code in a browser without needing to install or clone anything on their machine.
```
Comment (stevenbenitez) [inline README.md]: Use this project to challenge developers with relatively real-world examples related to RESTful APIs and frontend technologies. By loading this repository via [https://vscode.dev/github/heyitsgeo/peer-programming-challanges](https://vscode.dev/github/heyitsgeo/peer-programming-challanges) developers can edit code in a browser without needing to install or clone anything on their machine.

Comment (stevenbenitez) [reply README.md]: We will need to rename the repository too.

File: README.md [hunk 11]
```diff
@@ -1 +1,5 @@
...
+
+## Required extensions
+* [vue](https://marketplace.visualstudio.com/items?itemName=jcbuisson.vue)
```
Comment (stevenbenitez) [inline README.md]: What is this for? VS Code?

Comment (heyitsgeo) [reply README.md]: Ya without it vue templates are impossible to read in the online vscode tool.

Comment (stevenbenitez) [reply README.md]: Is this telling us that we need to install this or does the candidate need to install this? It's not clear to me what we are asking/instructing here.

**Pull request merged**